lib/versioning/docker/index.ts

```diff
@@ -65,16 +64,24 @@ class DockerVersioningApi extends GenericVersioningApi {
         return -1;
       }
       // alphabetic order
-      return parsed1.prerelease.localeCompare(parsed2.prerelease);
+      if (parsed1.prerelease && parsed2.prerelease) {
+        return parsed1.prerelease.localeCompare(parsed2.prerelease);
+      }
     }
     // equals
-    return parsed2.suffix.localeCompare(parsed1.suffix);
+    if (parsed2.suffix && parsed1.suffix) {
+      return parsed2.suffix.localeCompare(parsed1.suffix);
+    }
+
+    return 0;
```

Comment:
Snapshot change was caused by this line

Reply:
This is wrong, you need to check for empty string in suffix, so your change makes `1.2.3 === 1.2.3-alpine`. Same for prerelease